src/top/dashboard.rs

```diff
@@ -136,41 +131,52 @@ impl<'a> Widgets<'a> {
         let items = state.iter().map(|(_, r)| {
             let mut data = vec![r.name.clone(), r.kind.clone(), r.component_type.clone()];
 
-            // Build metric stats
-            let formatted_metrics = if self.opts.human_metrics {
-                [
-                    r.processed_events_total.human_format(),
-                    r.processed_events_throughput.human_format(),
-                    r.processed_bytes_total.human_format_bytes(),
-                    r.processed_bytes_throughput.human_format_bytes(),
-                    r.errors.human_format(),
-                ]
-            } else {
-                [
-                    r.processed_events_total.thousands_format(),
-                    r.processed_events_throughput.thousands_format(),
-                    r.processed_bytes_total.thousands_format(),
-                    r.processed_bytes_throughput.thousands_format(),
-                    r.errors.thousands_format(),
-                ]
-            };
+            let formatted_metrics = [
+                match r.processed_events_total {
+                    0 => "N/A".to_string(),
+                    v => format!(
+                        "{} ({}/s)",
+                        if self.opts.human_metrics {
+                            v.human_format()
+                        } else {
+                            v.thousands_format()
+                        },
+                        r.processed_events_throughput_sec.human_format()
+                    ),
+                },
+                match r.processed_bytes_total {
+                    0 => "N/A".to_string(),
+                    v => format!(
+                        "{} ({}/s)",
+                        if self.opts.human_metrics {
+                            v.human_format()
+                        } else {
+                            v.thousands_format()
+                        },
+                        r.processed_bytes_throughput_sec.human_format_bytes()
+                    ),
+                },
+                if self.opts.human_metrics {
+                    r.errors.human_format()
+                } else {
+                    r.errors.thousands_format()
+                },
+            ];
 
             data.extend_from_slice(&formatted_metrics);
             Row::StyledData(data.into_iter(), Style::default().fg(Color::White))
```

Comment:
Hey, off topic here, but noticed yesterday that  the CLion terminal does not like this `Color::White` here when you run `vector top`.  Shows up like below with the stats not readable.  "White" seems non-controversial and I would think it be handled everywhere, but on my machine the text is not visible...

<img width="1339" alt="Screen Shot 2020-11-26 at 4 14 04 PM" src="https://user-images.githubusercontent.com/4615775/100396488-72dbb000-3002-11eb-89ad-16e7630ac8c8.png">

Have you noticed this?  If you switch the CLion theme to MacOS light, the text goes to white...

<img width="1358" alt="Screen Shot 2020-11-26 at 4 15 18 PM" src="https://user-images.githubusercontent.com/4615775/100396549-ab7b8980-3002-11eb-8408-605d029ae8a3.png">

If you switch the line to just the "default" the text is visible again ("gray" also works and is used in other places in the dashboard).

```rs
Row::StyledData(data.into_iter(), Style::default())
```

Darcula theme:

<img width="1175" alt="Screen Shot 2020-11-26 at 4 20 09 PM" src="https://user-images.githubusercontent.com/4615775/100396775-a703a080-3003-11eb-924f-460970e30a18.png">

MacOS Light:

<img width="1194" alt="Screen Shot 2020-11-26 at 4 20 35 PM" src="https://user-images.githubusercontent.com/4615775/100396781-af5bdb80-3003-11eb-99d8-b854691c79d9.png">

Not sure what's different about the terminal in CLion to cause this.  Also, this part about dealing with the theme the person has chosen, was something I had not thought about before.  Thinking that TUI abstracts this away a bit and re-maps the colors somehow (?) , but haven't really looked at it.   Maybe it would be better to reports this issue in a ticket.

Reply:
Interesting! I'm not sure if tui can detect a custom theme, or whether that's just a term post-filter after rendering. In any case, it seems like we should probably drop `.fg(Color::White))` and side-step this.

Reply:
Looking at this further, I found this paragraph at https://perldoc.perl.org/Term::ANSIColor:

> Unfortunately, interpretation of colors 0 through 7 often depends on whether the emulator supports eight colors or sixteen colors. Emulators that only support eight colors (such as the Linux console) will display colors 0 through 7 with normal brightness and ignore colors 8 through 15, treating them the same as white. Emulators that support 16 colors, such as gnome-terminal, normally display colors 0 through 7 as dim or darker versions and colors 8 through 15 as normal brightness. On such emulators, the "normal" white (color 7) usually is shown as pale grey, requiring bright white (15) to be used to get a real white color. Bright black usually is a dark grey color, although some terminals display it as pure black. Some sixteen-color terminal emulators also treat normal yellow (color 3) as orange or brown, and bright yellow (color 11) as yellow.

I don't know whether that's relevant here, but crossterm interprets `Color::White` as:

```rust
Color::White => f.write_str("5;15")
```

Which would put it in the upper 16 colours. Wonder if that's why it's reverting to black?

Given that light themes are likely to be a popular option, I'm thinking opinionated colours may not be the best option, anyway. The use of MacOS Light highlighted that. Grey seems like a pretty universal/safe-ish default for asides, like the sample rate/quit instructions. Everything else, I'd probably just leave to the theme default.

Examples:

<img width="1054" alt="image" src="https://user-images.githubusercontent.com/719722/100435213-8be17100-3095-11eb-84f4-5166356f9a0e.png">

<img width="1054" alt="image" src="https://user-images.githubusercontent.com/719722/100435261-9b60ba00-3095-11eb-9b53-c51b75daeff8.png">

<img width="1054" alt="image" src="https://user-images.githubusercontent.com/719722/100435336-addaf380-3095-11eb-8774-0fea3e531592.png">

<img width="1057" alt="image" src="https://user-images.githubusercontent.com/719722/100435395-c64b0e00-3095-11eb-81ec-c6b5b5b829ee.png">